src/osd/OSD.cc

```diff
@@ -7458,6 +7467,13 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
 	do_restart = true;
       }
     }
+    if (osdmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS) !=
+	newmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
+      dout(10) << __func__ << " REQUIRE_LUMINOUS flag changed in " << newmap->get_epoch()
+	       << dendl;
+      clear_pg_stat_queue();
```

Comment:
@liewegas i don't think so. see https://github.com/ceph/ceph/pull/15073/commits/f71c18abbc4e6650ddbb0a72a25ea71697219e1f#diff-dfb9ddca0a3ee32b266623e8fa489626R2756 . please note, we are checking against the osdmap in OSD not `osdmap_ref` or `last_persisted_osdmap_ref`.

pg checks the osdmap flag before queuing themselves. imaging following sequence:

1. osdmap does not have `CEPH_OSDMAP_REQUIRE_LUMINOUS`, and pg enqueue itself 
1. osd consume an osdmap with `CEPH_OSDMAP_REQUIRE_LUMINOUS`, so the queue is cleared
1. pg does not enqueue itself anymore.